### DISCUSSION TITLE
Are non-Chrome versions planned?

### DISCUSSION THREAD

**Discussion by bershanskiy** [Maps4HTML/mapml-extension#33] [Q&A] [answered]
It appears that currently extension supports only Chromium browsers. Are there any plans for Firefox/Gecko or Safari versions? I believe that extension is sufficiently simple so that I can port it to Firefox in at most a few hours: make a simple build script (file copying, basically), polyfill `declarativeNetRequest` with `webRequest` on Firefox specifically, make existing tests run on Firefox, create tests for the code I add, etc. I'm not familiar with Safari at all and have no device to test on so I wouldn't be able to develop for it.

**Comment by prushforth** [comment, marked as answer]
That would be great! Re: safari I'm in the same position. Do you think you could use grunt for the build script? That's what we use in web-map-custom-element for builds...

**Reply by bershanskiy** [reply]
I ported `manifest.json` and `background.js` to Firefox and got Firefox to run all the necessary steps (change `Content-Type`, reload, and inject relevant CSS and JS), but map fails to appear. Oddly enough, console does not show any errors, just the log statements I added to see that JS is injected properly. I have a guess what the problem might be and I'll look into it this week.

**Reply by prushforth** [reply]
Do you have a reliable source of MapML?  The highlighted link on [this page](https://geogratis.gc.ca/mapml/en/cbmtile/cbmt/) should return `application/xml` which should work.

![image](https://user-images.githubusercontent.com/2437285/182444524-d59b1703-0a96-4397-95f0-8449a53ccfa1.png)

**Comment by prushforth** [comment]
I tried loading the firefox extension but couldn't.  Happy to help debug it once I get going.

**Reply by prushforth** [reply]
I realized that I had to build the extension with grunt in this branch, then I was able to load it in firefox.  I notice that it is using "quirks mode" when it reloads the document and I'm wondering if that's preventing custom elements from working.  How to avoid it loading / reloading in quirks mode?

**Reply by bershanskiy** [reply]
Hi, there are still a few bugs, I'm still working on it... I'll file a PR when it's ready!

**Reply by prushforth** [reply]
I can't wait!
